doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1814 @@
+
+These functions were previously used by libssl to set or get an encoded public
+key into/from an EVP_PKEY object. With OpenSSL 3.0 these are replaced by the more
+generic functions L<EVP_PKEY_set1_encoded_public_key(3)> and
+L<EVP_PKEY_get1_encoded_public_key(3)>.
+The old versions have been converted to deprecated macros that just call the
+new functions.
+
+=item EVP_PKEY_set1_engine and EVP_PKEY_get0_engine
+
+See L</Providers are a replacement for engines and low-level method overrides>.
+
+=item EVP_PKEY_set_alias_type
+
+This function has been removed. There is no replacement.
+See L</EVP_PKEY_set_alias_type() method has been removed>
+
+=item HMAC_Init_ex, HMAC_Update, HMAC_Final, HMAC_size
+
+=item HMAC_CTX_new, HMAC_CTX_free, HMAC_CTX_copy, HMAC_CTX_reset,
+HMAC_CTX_set_flags and HMAC_CTX_get_md
+
+See L</Deprecated low-level MAC functions>.
+
+=item i2d_DHparams, i2d_DHxparams
+
+=item i2d_DSAparams, i2d_DSAPrivateKey, i2d_DSAPrivateKey_bio,
+i2d_DSAPrivateKey_fp, i2d_DSA_PUBKEY, i2d_DSA_PUBKEY_bio, i2d_DSA_PUBKEY_fp and
+i2d_DSAPublicKey
+
+=item i2d_ECParameters, i2d_ECPrivateKey, i2d_ECPrivateKey_bio,
+i2d_ECPrivateKey_fp, i2d_EC_PUBKEY, i2d_EC_PUBKEY_bio, i2d_EC_PUBKEY_fp and
+i2o_ECPublicKey.
+
+=item i2d_RSAPrivateKey, i2d_RSAPrivateKey_bio, i2d_RSAPrivateKey_fp.
+i2d_RSA_PUBKEY, i2d_RSA_PUBKEY_bio, i2d_RSA_PUBKEY_fp, i2d_RSAPublicKey
+i2d_RSAPublicKey_bio and i2d_RSAPublicKey_fp
+
+See L</Deprecated low-level key reading and writing functions>
+and L<d2i_RSAPrivateKey(3)/Migration> 
+
+=item IDEA_encrypt, IDEA_set_decrypt_key, IDEA_set_encrypt_key,
+IDEA_cbc_encrypt, IDEA_cfb64_encrypt, IDEA_ecb_encrypt and IDEA_ofb64_encrypt
+
+=item MD2, MD2_Init, MD2_Update and MD2_Final
+
+=item MD4, MD4_Init, MD4_Update, MD4_Final and MD4_Transform
+
+=item MDC2, MDC2_Init, MDC2_Update and MDC2_Final
+
+See L</Deprecated low-level encryption functions>.
+IDEA, MD2, MD4 and MDC2 have been moved to the L<Legacy Provider|/Legacy Algorithms>.
+
+=item IDEA_options and MD2_options
+
+These functions return constant strings and are no longer supported.
+
+=item MD5, MD5_Init, MD5_Update, MD5_Final and MD5_Transform
+
+See L</Deprecated low-level encryption functions>.
+
+=item NCONF_WIN32
+
+This undocumented function has no replacement.
+See L<config(5)/HISTORY> for more details.
+
+=item OCSP_parse_url()
+
+Use L<OSSL_HTTP_parse_url(3)> instead.
+
+=item OCSP_REQ_CTX type and OCSP_REQ_CTX_*() methods.
+
+These methods were used to collect all necessary data to form a HTTP request,
+and to perform the HTTP transfer with that request.  With OpenSSL 3.0, the
+type is OSSL_HTTP_REQ_CTX, and the deprecated functions are replaced
+with OSSL_HTTP_REQ_CTX_*(). See L<OSSL_HTTP_REQ_CTX(3)> for additional details.
+
+=item OPENSSL_fork_child, OPENSSL_fork_parent and OPENSSL_fork_prepare
+
+There is no replacement for these functions. These pthread fork support methods
+were unused.
```

Comment:
Consider adding: _by OpenSSL_ to the end.  I'm sure some applications did use these.